src/ProjectOrigin.WalletSystem.Server/Models/Certificate.cs

```diff
@@ -1,6 +1,35 @@
 using System;
+using System.Collections.Generic;
 
 namespace ProjectOrigin.WalletSystem.Server.Models;
 
-public record Certificate(Guid Id,
-                          Guid RegistryId);
+public enum GranularCertificateType
+{
+    Invalid = 0,
```

Comment:
I do not think we need this type written here, even if it is exists on the types in protobuf, simply let c# throw an invalid cast exception.